### PR TITLE
Add support for vi mode indicator

### PR DESCRIPTION
**Pull request by arielserafini**

### CODE REVIEW

**Comment by will-stone** [comment]
I didn't even know this was possible 😄 I had a quick play and saw that this appears:

image

So I was wondering whether we even needed your pull but then I thought that your solution would look nicer with this prompt. My first suggestion would be to also change the colour of the (non-error) command mode to something else. Yellow would probably be a good colour (have a play but stick to the named colours).

My next suggestion is to change the names of the prompt icons, as you can see "default" doesn't really make sense now.

How about the following?

```
set __fish_prompt_char_insert '❯'
set __fish_prompt_char_command '❮'
```

File: README.md [hunk 1]
```diff
@@ -10,10 +10,12 @@ Forked from [Brandon Weiss's](http://github.com/brandonweiss/pure.fish/fork) ver
 
 [Download this file](https://raw.githubusercontent.com/will-stone/pure.fish/master/fish_prompt.fish) and save it as `~/.config/fish/functions/fish_prompt.fish`.
 
+If you want to use vi key bindings, create a new empty file and save it as `~/.config/fish/functions/fish_mode_prompt.fish` to disable the default mode indicator.
```
Comment (will-stone) [inline README.md]: I would say this comment is moot when we merge this PR. Unless you want to say something like:

```
If you'd like to use Vi key bindings, use the command `fish_vi_key_bindings` and `fish_default_key_bindings` to return to default.

**Comment by will-stone** [comment]
Also, visual mode should have a different indicator. Find a nice unicode V or V-like character.

File: README.md (hunk 1 above)
Comment (arielserafini) [reply README.md]: You need to have a no-op function (or empty file) for `fish_mode_prompt` to hide the default (ugly) `[I]` and `[N]` indicator. It's not great, but it's the only way to do it (to my knowledge).

**Comment by arielserafini** [comment]
Please see my reply to your code comment above. As far as a different character for visual mode, well, I'm basing this PR on the current Pure.zsh behavior and I'd rather keep the implementation as close as possible.

File: README.md (hunk 1 above)
Comment (will-stone) [reply README.md]: And you've included this in the install command. You've implied that creating `~/.config/fish/functions/fish_mode_prompt.fish` turns on Vi mode.

**Comment by will-stone** [comment]
Ah well, it's not meant to be exactly like Zsh's Pure (just based on). In fact I was thinking of renaming the project for this exact reason. Find a nice character for visual mode, please; Fish shows `[V]` for visual mode and I don't think this functionality should get lost.

**Comment by will-stone** [comment]
Hey @arielserafini I was wondering if you'd prefer to add this PR to Brandon's repo (where mine is forked from). He strives to be a little closer to the original Zsh and he's just merged my changes to simplify the file: https://github.com/brandonweiss/pure.fish

**Comment by will-stone** [comment]
@arielserafini Thank-you for your contribution. I will be closing this repo soon in favour of my own version of the Pure prompt with some additional tweaks. If you need a Pure prompt, please move this PR to Brandon's repo, or if you are happy with a tweaked version, please feel free to check out [impure](https://github.com/will-stone/impure)

**Pull request closed without merging**